formver.compiler-plugin/core/src/org/jetbrains/kotlin/formver/core/embeddings/callables/FullNamedFunctionSignature.kt

```diff
@@ -91,4 +90,17 @@ fun FullNamedFunctionSignature.toViperMethod(
     getPostconditions(returnVariable).pureToViper(toBuiltin = true),
     body,
     declarationSource.asPosition
-)
\ No newline at end of file
+)
+
+fun FullNamedFunctionSignature.toViperFunction(
+    body: Exp?,
+) = UserFunction(
+    name,
+    formalArgs.map { it.toLocalVarDecl() },
+    // TODO: Be explicit about the return types of functions instead of boxing them into a Ref
```

Comment:
This is optimistic at this stage; `Ref` is our standard type to pass things around, so this TODO would have to be everywhere if we wanted to mark all the places.